src/Features/Core/Portable/ExtractMethod/MethodExtractor.CodeGenerator.cs

```diff
@@ -83,97 +80,97 @@ protected CodeGenerator(InsertionPoint insertionPoint, SelectionResult selection
-                var callSiteDocument = await SemanticDocument.WithSyntaxRootAsync(root, cancellationToken).ConfigureAwait(false);
+                var newMethodDefinition = GenerateMethodDefinition(LocalFunction, cancellationToken);
+                var callSiteDocument = await InsertMethodAndUpdateCallSiteAsync(newMethodDefinition.Data, cancellationToken).ConfigureAwait(false);
+
+                // For nullable reference types, we can provide a better experience by reducing use of nullable
+                // reference types after a method is done being generated. If we can determine that the method never
+                // returns null, for example, then we can make the signature into a non-null reference type even though
+                // the original type was nullable. This allows our code generation to follow our recommendation of only
+                // using nullable when necessary. This is done after method generation instead of at analyzer time
+                // because it's purely based on the resulting code, which the generator can modify as needed. If return
+                // statements are added, the flow analysis could change to indicate something different. It's cleaner to
+                // rely on flow analysis of the final resulting code than to try and predict from the analyzer what will
+                // happen in the generator. 
+                var finalDocument = await UpdateMethodAfterGenerationAsync(callSiteDocument, newMethodDefinition, cancellationToken).ConfigureAwait(false);
+
+                return await CreateGeneratedCodeAsync(newMethodDefinition.Status, finalDocument, cancellationToken).ConfigureAwait(false);
+            }
 
-                var newCallSiteRoot = callSiteDocument.Root;
+            private Task<SemanticDocument> InsertMethodAndUpdateCallSiteAsync(
+                IMethodSymbol newMethodDefinition, CancellationToken cancellationToken)
+            {
+                var document = this.SemanticDocument.Document;
+                var codeGenerationService = document.GetLanguageService<ICodeGenerationService>();
 
-                var codeGenerationService = SemanticDocument.Document.GetLanguageService<ICodeGenerationService>();
-                var result = GenerateMethodDefinition(LocalFunction, cancellationToken);
+                return LocalFunction
+                    ? InsertLocalFunctionAndUpdateCallSiteAsync()
+                    : InsertNormalMethodAndUpdateCallSiteAsync();
 
-                SyntaxNode destination, newContainer;
-                if (LocalFunction)
+                async Task<SemanticDocument> InsertLocalFunctionAndUpdateCallSiteAsync()
                 {
-                    destination = InsertionPoint.With(callSiteDocument).GetContext();
+                    // First, update the callsite with the call to the new method.
+                    var outermostCallSiteContainer = GetOutermostCallSiteContainerToProcess(cancellationToken);
```

Comment:
note the use of outermostCallSiteContainer  here (Teh same outermostCallSiteContainer  we used when determining the insertion point of the local function).  By doing this we can get rid of hte prior code that had to check if we could no longer find the destinatino and bail out if htat happened.  this is also what enabled those failure tests to start succeeding.